wurst/objects/items/Armory/AxesDefinition.wurst

```diff
@@ -5,116 +5,174 @@ import AbilityObjEditing
-        this.setAreaofEffect(1, aoe)
-        this.setManaCost(1, 10)
-        this.setDurationHero(1, durationHero)
-        this.setDurationNormal(1, durationNormal)
-        this.setCastRange(1, castRange)
-        this.setCooldown(1, cooldown)
-        this.setEditorSuffix("(Wurst)")
-
-class Purge extends AbilityDefinitionPurgeCreep
-    construct(int abilId, string name, real durationHero, real durationNormal, real cooldown, real castRange)
-        super(abilId)
-        this.setName(name)
-        this.setManaCost(1, 0)
-        this.setDurationHero(1, durationHero)
-        this.setDurationNormal(1, durationNormal)
-        this.setCastRange(1, castRange)
-        this.setCooldown(1, cooldown)
-        this.setEditorSuffix("(Wurst)")
-
-@compiletime function createAxes()
-    new AxeDamageBonusAbility(ABIL_DMG_STONE , "Stone" , DMG_STONE)
-    new AxeDamageBonusAbility(ABIL_DMG_IRON  , "Iron"  , DMG_IRON)
-    new AxeDamageBonusAbility(ABIL_DMG_STEEL , "Steel" , DMG_STEEL)
-    new AxeDamageBonusAbility(ABIL_DMG_MASHER, "Masher", DMG_MASHER)
-    new AxeDamageBonusAbility(ABIL_DMG_BATTLE, "Battle", DMG_BATTLE)
-
-    new Silence(ABIL_SILENCE, "Mage Masher Silence", SILENCE_DURATION_HERO, SILENCE_DURATION_NORMAL, SILENCE_COOLDOWN, SILENCE_CAST_RANGE, SILENCE_AOE)
-    new Purge(ABIL_PURGE, "Battle Axe Purge", PURGE_DURATION_HERO, PURGE_DURATION_NORMAL, PURGE_COOLDOWN, PURGE_CAST_RANGE)
-
-    new Axe(ITEM_STONE_AXE   , Icons.bTNOrcMeleeUpOne              , ST_MODEL , "Stone Axe"   , "A", TT_COMMON , DMG_STONE , 12, asList(255, 255 , 255 ), ABIL_STONE)
-    new Axe(ITEM_IRON_AXE    , Icons.bTNOrcMeleeUpTwo              , IA_MODEL , "Iron Axe"    , "S", TT_COMMON , DMG_IRON  , 25, asList(255, 255 , 255 ), ABIL_IRON)
-    new Axe(ITEM_STEEL_AXE   , Icons.bTNSpiritWalkerAdeptTraining  , SA_MODEL , "Steel Axe"   , "Q", TT_COMMON , DMG_STEEL , 48, asList(120, 140 , 255 ), ABIL_STEEL)
-
-    new UsableAxe(ITEM_BATTLE_AXE  , Icons.bTNOrcMeleeUpThree            , BA_MODEL , "Battle Axe"  , "A", TT_BA     , DMG_BATTLE, 60, asList(255, 255 , 255 ), ABIL_BATTLE)
-    new UsableAxe(ITEM_MAGE_MASHER , Icons.bTNSpiritWalkerMasterTraining , MM_MODEL , "Mage Masher" , "A", TT_MM     , DMG_MASHER, 40, asList(255, 65  , 65  ), ABIL_MASHER)
+let TT_BA = "A two handed axe. " + TT_COMMON + " Can cast purge, removes all buffs from a target unit, and slows its movement speed." + formatCooldown(ABIL_PURGE) + formatDuration(ABIL_PURGE)
+
+ // Only purpose of this function is distinction, not sure about cooldown group, should battle axe and masher share cooldown?
+function createAxe(int axeId) returns ItemDefinition
+    return createBasicItem(axeId)
+        ..setCooldownGroup("AIas")
+
+@compiletime function createStoneAxe() returns ItemDefinition
+    return createAxe(ITEM_STONE_AXE)
+        ..setInterfaceIcon(Icons.bTNOrcMeleeUpOne)
+        ..setModelUsed(LocalItems.stoneAxe)
+        ..setName("Stone Axe")
+        ..setTooltipBasic("Trade for Stone Axe")
```

Comment:
Ahead of doing this *everywhere*, we should create a utility that updates the name and basic tooltip in this pattern (except it should say "Purchase").

Reply:
No clue on how to do this.

Reply:
Define a new method in a local file, e.g.

    public function ItemDefinition.setNameEnhance(name)
        this
            ..setName(name)
            ..setTooltipBasic("Purchase " + name)

You could import ItemDefinition publicly to simplify imports too and then just prefer this import everywhere. Alternatively the tooltip generator could do this for you potentially, which is the better way if it is possible.

Reply:
Item's don't support tooltip generation, so you'd need to use the method approach. I had thoughts of doing something for generating kits and buildings within a single shared definition wrapper, so we can discuss this more if it's not clear.

Reply:
Cool, learnt something new.